src/types/CanonicalField.ts

```diff
@@ -0,0 +1,49 @@
+import { ajv } from '../ajv';
+import type { JSONSchemaType } from 'ajv';
+
+export interface CanonicalField {
+  id: number;
+  label: string;
+  shortCode: string;
+  dataType: string;
+  createdAt: Date;
+}
+
+export const canonicalFieldSchema: JSONSchemaType<CanonicalField> = {
+  type: 'object',
+  properties: {
+    id: {
+      type: 'integer',
+    },
+    label: {
+      type: 'string',
+    },
+    shortCode: {
+      type: 'string',
+    },
+    dataType: {
+      type: 'string',
+    },
+    createdAt: {
+      type: 'object',
+      required: [],
+      instanceof: 'Date',
+    },
+  },
+  required: [
+    'id',
+    'label',
+    'shortCode',
+    'dataType',
+    'createdAt',
+  ],
+};
+
+export const isCanonicalField = ajv.compile(canonicalFieldSchema);
+
+const canonicalFieldArraySchema: JSONSchemaType<CanonicalField[]> = {
+  type: 'array',
+  items: canonicalFieldSchema,
+};
+
+export const isCanonicalFieldArray = ajv.compile(canonicalFieldArraySchema);
```

Comment:
This looks great and I am comfortable with the more targeted utility types being used in the context where they are needed. I guess I still have a feeling that there will be a relatively common shape for exactly one other type per entity. We can cross that bridge when we come to it, though, if it comes.